### DISCUSSION TITLE
How to install language support offline?

### DISCUSSION THREAD

**Discussion by futurehome** [nvim-treesitter/nvim-treesitter#4423] [General]
I want to install a language support, but I can't access the internet from my working computer. I can download the zip file, is there any way to install the zip file offline?

**Comment by clason** [comment]
You can change the `parsers` table field to point at a local directory where you have cloned the repo.

**Reply by qlibp** [reply]
one more question on this, if I wanna add cpp support, as it has a npm_require field, how should I accomplish it if I got no internet to download npm package?

I go further, and found out that its grammar.js requires treesitter-c, so if I place treessitter-c repo and treesitter-cpp repo locally, could treesitter cpp module be compiled successfully?

**Reply by theHamsta** [reply]
You can use `:TSInstall`, it will rely on the generated `src/parser.c` and not run any `npm`. Only `:TSInstallFromGrammar`, will regenerate the grammar and for that rerun `npm install`.

**Reply by qlibp** [reply]
I will have a try then, thx for the reply!
